packages/webapp/hooks/useSharedByToast.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useRef } from 'react';
+import { useRouter } from 'next/router';
+import Link from 'next/link';
+import { useAuthContext } from '@dailydotdev/shared/src/contexts/AuthContext';
+import { useToastNotification } from '@dailydotdev/shared/src/hooks/useToastNotification';
+import { useContentPreferenceStatusQuery } from '@dailydotdev/shared/src/hooks/contentPreference/useContentPreferenceStatusQuery';
+import {
+  ContentPreferenceStatus,
+  ContentPreferenceType,
+} from '@dailydotdev/shared/src/graphql/contentPreference';
+import { useUserShortByIdQuery } from '@dailydotdev/shared/src/hooks/user/useUserShortByIdQuery';
+import { useContentPreference } from '@dailydotdev/shared/src/hooks/contentPreference/useContentPreference';
+import { ButtonSize } from '@dailydotdev/shared/src/components/buttons/common';
+import {
+  ProfileImageSize,
+  ProfilePicture,
+} from '@dailydotdev/shared/src/components/ProfilePicture';
+import { Button } from '@dailydotdev/shared/src/components/buttons/Button';
+import {
+  Typography,
+  TypographyTag,
+  TypographyType,
+} from '@dailydotdev/shared/src/components/typography/Typography';
+
+const useSharedByToast = (): void => {
+  const hasShownToast = useRef(false);
+  const { user: currentUser, isAuthReady } = useAuthContext();
+  const { query } = useRouter();
+  const userId = (query?.userid as string) || null;
+  const isSameUser = !!userId && userId === currentUser?.id;
+  const { data: contentPreference, isPending } =
+    useContentPreferenceStatusQuery({
+      id: userId,
+      entity: ContentPreferenceType.User,
+    });
+  const { data: user } = useUserShortByIdQuery({ id: userId });
+  const { follow } = useContentPreference({ showToastOnSuccess: true });
+  const { displayToast, dismissToast } = useToastNotification();
+  const isDataReady = isAuthReady || (currentUser && !isPending);
+
+  useEffect(() => {
+    if (
+      !user ||
+      isSameUser ||
+      !isDataReady ||
+      hasShownToast.current ||
+      contentPreference?.status === ContentPreferenceStatus.Blocked
+    ) {
+      return undefined;
+    }
+
+    const showFollow =
+      !!currentUser &&
+      contentPreference?.status !== ContentPreferenceStatus.Follow;
```

Comment:
I think this condition is not good since if I see toast from Nimrod for example I see "Follow" and I am already following. I think you need to check if follow or subscribe:
```suggestion
    const showFollow =
      !!currentUser &&
      ![ContentPreferenceStatus.Follow, ContentPreferenceStatus.Subscribed].includes(contentPreference?.status);
```